trunk/SUAVE/Plots/Geometry_Plots/plot_vehicle.py

```diff
@@ -39,286 +40,582 @@ def plot_vehicle(vehicle, save_figure = False, plot_control_points = True, save_
+
+    Assumptions: 
+    None
+
+    Source:   
+    None
     
- 
-    for propulsor in vehicle.propulsors:    
-        if ('propeller' in propulsor.keys()) or ('rotor' in propulsor.keys()): 
-            try:
-                prop = propulsor.propeller
-            except:
-                prop = propulsor.rotor
+    Inputs:     
+    
+    Properties Used:
+    N/A
+    """      
+     
+    num_fus_segs = len(fus_pts[:,0,0])
+    tesselation  = len(fus_pts[0,:,0]) 
+    for i_seg in range(num_fus_segs-1):
+        for i_tes in range(tesselation-1):
+            X = [fus_pts[i_seg  ,i_tes  ,0],
+                 fus_pts[i_seg  ,i_tes+1,0],
+                 fus_pts[i_seg+1,i_tes+1,0],
+                 fus_pts[i_seg+1,i_tes  ,0]]
+            Y = [fus_pts[i_seg  ,i_tes  ,1],
+                 fus_pts[i_seg  ,i_tes+1,1],
+                 fus_pts[i_seg+1,i_tes+1,1],
+                 fus_pts[i_seg+1,i_tes  ,1]]
+            Z = [fus_pts[i_seg  ,i_tes  ,2],
+                 fus_pts[i_seg  ,i_tes+1,2],
+                 fus_pts[i_seg+1,i_tes+1,2],
+                 fus_pts[i_seg+1,i_tes  ,2]]                 
+            verts = [list(zip(X, Y, Z))]
+            collection = Poly3DCollection(verts)
+            collection.set_facecolor(face_color)
+            collection.set_edgecolor(edge_color) 
+            collection.set_alpha(alpha)
+            axes.add_collection3d(collection)  
+    
+    return 
+
+
+def plot_propulsor(axes,VD,propulsor,propulsor_face_color,propulsor_edge_color,propulsor_alpha):  
+    """ This plots a 3D surface of the fuselage
+
+    Assumptions: 
+    None
+
+    Source:   
+    None
+    
+    Inputs:   
+    VD                   - vortex distribution    
+    
+    Properties Used:
+    N/A
+    """          
+    
+    if ('propeller' in propulsor.keys()):  
+        prop = propulsor.propeller 
+        try:
+            propulsor.thrust_angle = propulsor.propeller_thrust_angle     
+        except:
+            pass 
+        
+        # Generate And Plot Propeller/Rotor Geoemtry   
+        plot_propeller_geometry(axes,prop,propulsor,'propeller')
+
+        # Generate Nacelle Geoemtry
+        nac_geo = generate_nacelle_points(VD,propulsor,'propeller')
+        
+        # Plot Nacel Geometry 
+        plot_nacelle(axes,nac_geo,propulsor_face_color,propulsor_edge_color,propulsor_alpha)
+           
+    if ('rotor' in propulsor.keys()):  
+        rot = propulsor.rotor   
+        try:
+            propulsor.thrust_angle = propulsor.rotor_thrust_angle     
+        except:
+            pass
+        
+        # Generate And Plot Propeller/Rotor Geoemtry   
+        plot_propeller_geometry(axes,rot,propulsor,'rotor')
+    
+        # Generate Nacelle Geoemtry 
+        nac_geo = generate_nacelle_points(VD,propulsor,'rotor')
+        
+        # Plot Nacel Geometry 
+        plot_nacelle(axes,nac_geo,propulsor_face_color,propulsor_edge_color,propulsor_alpha) 
+
+    
+    elif 'turbofan' ==  propulsor.tag: 
```

Comment:
rather than checking tags check for isinstance()

Reply:
Changed to isinstance()